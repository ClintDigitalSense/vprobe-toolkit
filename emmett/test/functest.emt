void function(int a, int b);
void noArgs();
void vafunc(int a, ...);
int returnsInt();
string returnsStr(int snap, ...);

int foo;
string nameStr;

string
name(string str, int arg)
{
   sprintf(str, "fooble: %d\n", arg);
   return str;
}  

void
voidFunc(int arg)
{
  printf("void function with an arg!\n");
}

void
voidFunc2()
{
   printf("void function with no args\n");
}

MyFunc {
   foo = returnsInt();
   function(foo, 2*1+3*bar(13));
   vafunc(foo);
   vafunc(foo, "args are fun!\n");
   vafunc(foo, "args are fun",
          "-ctional\n");
   name(nameStr, 12);
   printf("%s\n", nameStr);
   voidFunc(1);
   voidFunc2();
}
