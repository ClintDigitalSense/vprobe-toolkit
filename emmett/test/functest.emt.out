(version 0.2)
(definteger foo)
(defstring nameStr)
(defun name(str arg )( do  ( sprintf    str  "fooble: %d\n"  arg ))str)(defun voidFunc(arg )( do  ( printf    "void function with an arg!\n" )))(defun voidFunc2()( do  ( printf    "void function with no args\n" )))
 ;;;;;
(vprobe MyFunc ( do  ( setint foo    (returnsInt))( function    foo   (+ 2   (* 3   (bar 13 ) ) ) )( vafunc    foo )( vafunc    foo  "args are fun!\n" )( vafunc    foo  "args are fun"  "-ctional\n" )( name    nameStr  12 )( printf    "%s\n"  nameStr )( voidFunc    1 )( voidFunc2   )))